### PR TITLE
Add trace level debugging

### PR DESCRIPTION
**Pull request by DonJayamanne**
TODO: rename traceTrace to logTrace or other in a separate PR,
Else that will be too large,
The idea is `traceVerbose` will be renamed to `logVerbose` and the like.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1461,7 +1461,16 @@
                         "error",
                         "warn",
                         "info",
-                        "debug"
+                        "debug",
+                        "trace"
```
Comment (DonJayamanne) [inline package.json:1465]: New item

File: src/extension.node.ts [hunk 2]
```diff
@@ -168,30 +178,6 @@ function activateUnsafe(context: IExtensionContext): [IExtensionApi, Promise<voi
     }
 }
 
-/////////////////////////////
-// error handling
-
-async function activateComponents(
```
Comment (DonJayamanne) [inline src/extension.node.ts:174]: Moved duplicate code into extension.common.ts

File: src/platform/logging/outputChannelLogger.ts [hunk 3]
```diff
@@ -31,23 +31,23 @@ export class OutputChannelLogger implements ILogger {
         return logMessage;
     }
 
-    public traceLog(message: string, ...data: Arguments): void {
```
Comment (DonJayamanne) [inline src/platform/logging/outputChannelLogger.ts:34]: Dead code

File: src/platform/common/configSettings.ts [hunk 4]
```diff
@@ -41,7 +38,6 @@ export class JupyterSettings implements IWatchableJupyterSettings {
 
     private static jupyterSettings: Map<string, JupyterSettings> = new Map<string, JupyterSettings>();
     public experiments!: IExperiments;
-    public logging: ILoggingSettings = { level: LogLevel.Error };
```
Comment (DonJayamanne) [inline src/platform/common/configSettings.ts:44]: More layers gone

**Review by roblourens — APPROVED** [review]

**Pull request merged**